### DISCUSSION TITLE
Error 'Lesson 6': While deploying DeploySimpleStorage.sol

### DISCUSSION THREAD

**Discussion by deepakA18** [Cyfrin/foundry-full-course-cu#318] [General]
I'm facing an error while deploying the DeploySimpleStorage.sol using this command:
`   $ forge script script/DeploySimpleStorage.sol --rpc-url http://127.0.0.1:8545 `

The error says: 
error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)

**Comment by shikhar229169** [comment]
Have you started the anvil chain?

**Reply by deepakA18** [reply]
I faced difficulty in importing the account with fake currency by manually adding the network , though I was copy pasting the rpc url and chain ID it was saying rpc url is invalid.
I entered- 
RPC-url: - http://127.0.0.1:8545
Chain ID - 31337

**Comment by basefee** [comment]
Just spin up your Anvil local chain.
Try using the command without --rpc-url, for anvil you do not need to mention the rpcurl.
As it will by default run in anvil.

**Comment by Aishwaria8786** [comment]
I guess you need to type anvil in the terminal.

**Reply by basefee** [reply]
Yes we need to run anvil in a terminal. Then use the command

**Reply by cromewar** [reply]
Most likely the correct answer @deepakA18

**Reply by ghost** [reply]
getting the same error but i'm using ganache so what should i do ?
command:
'''
forge create SimpleStorage --rpc-url http://127.0.0.1:7545 --interactive
'''
Error:
'''
[⠒] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
'''

**Reply by basefee** [reply]
Recheck your RPC_URL, it seems the entered url is incorrect
Or check if ganache is running.

**Reply by deepakA18** [reply]
@garv-aga @anjalisgit  rpc url of ganache http://127.0.0.1:7545/ is correct.
The issue is the chain is not started.
That is why it is facing problem to connect.

**Reply by ghost** [reply]
It's resolved now, thanks
Needed to change the server to WSL from ganache settings and change the url

**Reply by danMendes-bossanova** [reply]
I'm new on this and get this exactly same Error right now. How did you solve this. Can you explain to me?

**Reply by Ragin-Bull** [reply]
@danMendes-bossanova You need to open up Ganache, then move to settings followed by server and then change your host name to WSL (If you're operating on Windows) and then copy the new changed URL and run the previous command. It should work now.

**Reply by visnupb** [reply]
```
vichu@DESKTOP:~/Foundry/SimpleStorage$ forge create SimpleStorage --rpc-url http://172.19.
160.1:7545
[⠃] Compiling...No files changed, compilation skipped
[⠊] Compiling...
```
 My Ganache is running and no error..Its just stuck in here. Please help if someone finds a solution .
 
 Thank you!!

**Reply by Vancelott** [reply]
@Ragin-Bull My issue was resolved thanks to your tip, thank you!

Here's the full error I was running into when I tried to change my rpc url:

```
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)
```

By the way, if you are struggling with the error, I suggest you don't worry about it and just watch the video, as Patrick switches to Anvil after showing that the contract is successfully deployed via Ganache.

**Comment by Vishal-coding** [comment]
@deepakA18  $ forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545 can try this renaming your script file this really suits if you had already solved this happy to hear.

**Comment by christian1415** [comment]
For those who still don't know what's the issue: You need to have anvil running, don't CTRL-C!
Open a new terminal and type 'anvil' and switch back to 'bash'.
Now you can run the command: ` forge create SimpleStorage --interactive`
Hope this helps someone.

**Reply by solidityenj0yer** [reply]
Worked for me this way. I had so much trouble fixing it thank you !! 😆

**Reply by shadowalpha197** [reply]
Yes, this worked for me too. Thanks!

**Comment by mimiprosper** [comment]
Please make sure Anvil is running. Anvil is running on http://127.0.0.1:8545. For better result open two terminals, on for running Anvil and the other for running the scripts

**Reply by EngrPips** [reply]
Indeed

**Reply by 0xKann** [reply]
Thanks guys I fixed it . I just ran 2 terminals from the first one  I just ran anvil . From second one I ran my commands forge create  src/Counter.sol:Counter --interactive

**Reply by EngrPips** [reply]
Interesting.
